src/components/pages/MentorMenteeMatching/MentorOptions.js

```diff
@@ -1,21 +1,84 @@
 import { Select } from 'antd';
+import { useEffect, useState } from 'react';
+import axiosWithAuth from '../../../utils/axiosWithAuth';
+
+export const Selected = (edit, setMentor, mentor, data, name, nameVal) => {
+  /**
+   * Owner: Khaleel Musleh
+   * Props Being Passed:
+   * Props: mentor parameter passed all the way from MentorMenteeMatching -> MentorMenteeInfo -> MentorsCard -> MentorsOptions
+   */
+
+  const [mentors, setMentors] = useState([]);
+
+  /**
+   * Owner: Khaleel Musleh
+   * Selection variable has all the HTML elements to pass to the list of selections for each user, if a user has no mentors then it will return no data.
+   *
+   */
+  let selections = [];
+
+  const { Option } = Select;
+  /**
+   * Owner: Khaleel Musleh
+   * GET mentors based on the mentor_id found in the mentor props, it passes all the ID's into the GET function below and gets the mentors of each user seperately.
+   * setMentor gets the data of the mentor information then its passed below to selection
+   */
+  useEffect(() => {
+    axiosWithAuth()
+      .get(`/assignments/mentee/${mentor.key}`)
+      .then(response => {
+        setMentors(response.data);
+      });
+  }, []);
```

Comment:
Good job not forgetting that dependency array!

Reply:
thank you